src/Transport/Config/AzureServiceBusTransportInfrastructure.cs

```diff
@@ -14,6 +15,11 @@ class AzureServiceBusTransportInfrastructure : TransportInfrastructure
         ITopology topology;
         SatelliteTransportAddressCollection satelliteTransportAddresses;
 
+        public override Task Stop()
+        {
+            return ((IStoppableTopology) topology).Stop();
```

Comment:
why not just resolve the factory lifecycle manager here and stop all? Why introduce a new interface and then implement it twice?